File: Py/demo/app-simple.py
from shiny import App, ui, render

# Define UI
app_ui = ui.page_fluid(
    ui.h1("Simple Shiny App"),
    # Card with slider
    ui.card(
        ui.card_header("Welcome"),
        ui.p("This is a basic Shiny for Python application."),
        # Input: Slider
        ui.input_text_area("text", "What's your question?"),
        # Output: echo text back
        ui.output_text("echo"),
    ),
)


# Define server
def server(input, output, session):
    @render.text
    def echo():
        return f"You asked: {input.text()}"


# Create app
app = App(app_ui, server)

File: R/demo/app-simple.R
library(shiny)
library(bslib)

ui <- page_fluid(
  theme = bs_theme(bootswatch = "minty"),

  # App title
  h1("DeckCheck"),

  # Create a card
  card(
    card_header("Welcome"),
    p("This is a simple Shiny app using bslib for layout and styling."),

    # Input: Slider
    textAreaInput("text", "What's your question?"),

    # Output: echo the text back
    textOutput("echo")
  )
)

server <- function(input, output, session) {
  # Reactive output
  output$echo <- renderText({
    paste("You asked:", input$text)
  })
}

shinyApp(ui, server)
